geopandas/geodataframe.py

```diff
@@ -546,6 +546,9 @@ def from_features(cls, features, crs=None, columns=None):
             Optionally specify the column names to include in the output frame.
             This does not overwrite the property names of the input, but can
             ensure a consistent output format.
+        id_as_index: bool (optional)
```

Comment:
If this is to be merged, think this should specify whether it's using `properties['id']` or `feature['id']`... on first brush this is not clear to me. 

I think the option would be better implemented as `index=None/str/collection` would be clearer, mimicking the `pandas.DataFrame` index argument? If `index='id'`, then we use column `id` in features. If `index=None`, then we try `feature['id']`, otherwise use a `RangeIndex`. 

I don't clearly see a way to manage the synchronization (or lack thereof) between `feature['id']` and `property['id']`, especially if we write the GeoDataFrame's index back as the `id`.